### DISCUSSION TITLE
How to install on a dynamic partition system?

### DISCUSSION THREAD

**Discussion by user-grinch** [phhusson/treble_experimentations#1896] [Q&A] [answered]
I'm trying to install android 11 GSI on a dynamic partition device (Andriod 10). I've tried but it always ended with a bootloop. Did anyone have success with this? List the steps below, thanks.

Device: Poco M2 Shiva

**Comment by phhusson** [comment, marked as answer]
0. Get an up-to-date fastboot on your computer (fastboot —version should
give version >= 29)
1. Get vbmeta.img from
https://dl.google.com/developers/android/qt/images/gsi/vbmeta.img
2. Get A/B GSI (I'm guessing you need ARM64), don't forget to uncompress it
3. From running Android, do adb reboot bootloader
4. fastboot flash vbmeta vbmeta.img
5. fastboot reboot fastboot
6. fastboot flash system system-xxxx.img
6bis. If fastboot tells you there isn't enough place, do fastboot
delete-logical-partition product, fastboot delete-logical-partition
product_a, fastboot delete-logical-partition product_b and run the fastboot
flash command again
7. On your phone, the screen should have a button "go back to recovery",
select it, then select "factory reset / wipe data"
8. Reboot and enjoy

**Reply by user-grinch** [reply]
Thanks, it worked, the issue was I was using the "Treble check" app which was telling me my device was a only but it's a/b. Thanks for correcting that. Btw if something doesn't work, how can I report it?

**Reply by itayalush** [reply]
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')
FAILED (remote: 'No such FUCKING file or directory')

**Reply by devrieshh** [reply]
after done all the steps,it says all is installed no errors,doing fastboot -w  it uses 0.03sec ,and just restart to the page with the notion that there is unlocked bootloader, it wil restart in 5sec.....
factory reset does says it cant restart,or something simular...

i think the gsi isnt written...
vndk 31
Cubot king kong mini3

**Comment by phhusson** [comment]
You open a ticket in this bug tracker, or you come on the telegram group

11 juin 2021 15:12:46 Grinch_ ***@***.***>:

**Comment by itayalush** [comment]
FAILED (remote: 'No such FUCKING file or directory')

**Comment by devrieshh** [comment]
get the treble info app,to see what system you need A64 or amd64.
do all the steps told some items ago,but 5 fastboot -w
                                                                6 fastboot reboot fastboot
                                                                7./fastboot flash system `your choosen system .img`
be sure  to extract the .img xz file 
in terminal    unxz LeOS-VNDK-a64-bvN-19.1.img.xz  (for example)

sometimes on certain devices you have to use ./fastboot   instead of   fastboot
